### PR TITLE
#11 object menu unit tests

### PR DESCRIPTION
**Pull request by MariaR001**
- Implemented basic unit, component and api tests
- Implemented collaboration between react, vite and cypress for coverage reports 
- Added vitest coverage report action -> moved to next iteration

**Make sure to run the following commands under UInnovateApp
*Make sure to run: npm install and npm install cypress -D

test the following:
- npm run cy:run-component:
image

 Also, creates a coverage folder where you find the collab between react, vite and cypress (UInnovate\UInnovateApp\coverage -> index.html)
image

- npx vitest --coverage: displays coverage report with vitest
image

- npx cypress open -> then click on Component testing -> App you should see this:
image

**need to run the application at the same time for e2e (npm run dev)
**if there's a router error/warning just ignore for now as long as there's a check mark next to "should visit"
then go back, select e2e testing and select 'spec' you should see this:
image

if you select api you should see this:
image

### CODE REVIEW

**Comment by MariaR001** [comment]
making a seperate PR for coverage report action

File: README.md [hunk 1]
```diff
@@ -147,6 +147,14 @@ Our current coverage reports only support Vitest unit tests by running
 npm run coverage
 ```
 
+## Start vite server and run cypress
+
+To run the vite server and run the cypress application at the same time
+
+```bash
+npm run cy:e2e
```
Comment (lyubageorgieva) [inline README.md:155]: In a future commit, we could also update the table of available scripts to have this command.

**Review by lyubageorgieva — DISMISSED** [review]
Very well documented PR! All the tests pass on my side. Great work Maria! 😊

**Review by lyubageorgieva — APPROVED** [review]
The issue regarding the /app not loading is fixed. Everything works on my end!

**Pull request merged**